magicctapipe/io/tests/test_io.py

```diff
@@ -19,619 +21,721 @@
+@pytest.mark.dependency(depends=["test_exist_dl1_stereo_mc"])
+class TestStereoMC:
+    def test_get_stereo_events_mc(self, gamma_stereo, p_stereo, config_gen):
+        """
+        Check on stereo data reading
+        """
 
-
-def test_load_mc_dl2_data_file(p_dl2, gamma_dl2):
-    """
-    Checks on default loading
-    """
-    dl2_mc = [p for p in gamma_dl2.glob("*")] + [p for p in p_dl2.glob("*")]
-    for file in dl2_mc:
-        data, point, _ = load_mc_dl2_data_file(
-            str(file), "width>0", "software", "simple"
+        stereo_mc = (
+            [p for p in gamma_stereo[0].glob("*")]
+            + [p for p in gamma_stereo[1].glob("*")]
+            + [p for p in p_stereo[0].glob("*")]
+            + [p for p in p_stereo[1].glob("*")]
         )
-        assert "pointing_alt" in data.colnames
-        assert "theta" in data.colnames
-        assert "true_source_fov_offset" in data.colnames
-        assert data["true_energy"].unit == "TeV"
-        assert point[0] >= 0
-        assert point[0] <= 90
 
-
-def test_load_mc_dl2_data_file_cut(p_dl2, gamma_dl2):
-    """
-    Check on quality cuts
-    """
-    dl2_mc = [p for p in gamma_dl2.glob("*")] + [p for p in p_dl2.glob("*")]
-    for file in dl2_mc:
-        data, _, _ = load_mc_dl2_data_file(
-            str(file), "gammaness>0.1", "software", "simple"
+        for file in stereo_mc:
+            event_data = pd.read_hdf(str(file), key="events/parameters")
+            event_data.set_index(["obs_id", "event_id", "tel_id"], inplace=True)
+            event_data.sort_index(inplace=True)
+            data = get_stereo_events(event_data, config_gen)
+            assert np.all(data["multiplicity"] > 1)
+            assert np.all(data["combo_type"] >= 0)
+
+    def test_get_stereo_events_mc_cut(self, gamma_stereo, p_stereo, config_gen):
+        """
+        Check on quality cuts
+        """
+        stereo_mc = (
+            [p for p in gamma_stereo[0].glob("*")]
+            + [p for p in gamma_stereo[1].glob("*")]
+            + [p for p in p_stereo[0].glob("*")]
+            + [p for p in p_stereo[1].glob("*")]
         )
-        assert np.all(data["gammaness"] > 0.1)
-
-
-def test_load_mc_dl2_data_file_opt(p_dl2, gamma_dl2):
-    """
-    Check on event_type
-    """
-    dl2_mc = [p for p in gamma_dl2.glob("*")] + [p for p in p_dl2.glob("*")]
-    for file in dl2_mc:
-        data_s, _, _ = load_mc_dl2_data_file(str(file), "width>0", "software", "simple")
-        assert np.all(data_s["combo_type"] < 3)
+        for file in stereo_mc:
+            event_data = pd.read_hdf(str(file), key="events/parameters")
+            event_data.set_index(["obs_id", "event_id", "tel_id"], inplace=True)
+            event_data.sort_index(inplace=True)
+            data = get_stereo_events(event_data, config_gen, "intensity>50")
+            assert np.all(data["intensity"] > 50)
+
+    def test_load_train_data_files_p(self, p_stereo):
+        """
+        Check dictionary
```

Comment:
maybe write explicitely that you check the dictorionary of the combo_type naming

Reply:
done